processor/src/host/advice/mod.rs

```diff
@@ -346,3 +346,44 @@ impl From<AdviceInputs> for AdviceProvider {
         Self { stack, map, store }
     }
 }
+
+impl AdviceProviderInterface for AdviceProvider {
+    #[inline(always)]
+    fn pop_stack(&mut self) -> Result<Felt, AdviceError> {
+        self.pop_stack()
+    }
+
+    #[inline(always)]
+    fn pop_stack_word(&mut self) -> Result<Word, AdviceError> {
+        self.pop_stack_word()
+    }
+
+    #[inline(always)]
+    fn pop_stack_dword(&mut self) -> Result<[Word; 2], AdviceError> {
+        self.pop_stack_dword()
+    }
+
+    #[inline(always)]
+    fn get_merkle_path(
+        &self,
+        root: Word,
+        depth: &Felt,
+        index: &Felt,
+    ) -> Result<MerklePath, AdviceError> {
+        self.get_merkle_path(root, depth, index)
+    }
+
+    #[inline(always)]
+    fn update_merkle_node(
+        &mut self,
+        root: Word,
+        depth: &Felt,
+        index: &Felt,
+        value: Word,
+    ) -> Result<(MerklePath, Word), AdviceError> {
+        self.update_merkle_node(root, depth, index, value)
+
+        // TODO(plafer): return error if depth is invalid, equivalent to:
+        // assert_eq!(path.len(), depth.as_int() as usize);
```

Comment:
Q: is this the right place for this?

Reply:
Yes, although currently unintuitive. In the main body of `op_mrupdate()`, we currently can't assume that we have a valid `MerklePath` (e.g. see how this method is implemented for `AdviceReplay`). So if we were to run that check there, it would always fail in the fast processor.

I wanted to investigate using `Option<MerklePath>` instead as the return type to more properly model this in a follow-up PR, but I might just do it in this one because I agree that it's confusing.

Reply:
Added a commit that uses `Option<MerklePath>`, and moves the check outside